common/djangoapps/third_party_auth/views.py

```diff
@@ -101,20 +104,37 @@ def get(self, request):
         """
         Render a form to collect user input about data sharing consent
         """
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
         running_pipeline = get_running_pipeline(request)
         if running_pipeline:
             request.session['quarantined_module'] = 'third_party_auth.'
             current_provider = Registry.get_from_pipeline(running_pipeline)
             if current_provider:
                 name = current_provider.name
+                data_sharing_consent = current_provider.data_sharing_consent
             else:
                 raise Http404
         else:
             raise Http404
         data = {
-            'platform_name': configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
+            'platform_name': platform_name,
             'sso_provider': name,
+            'data_sharing_consent': data_sharing_consent,
             'csrftoken': csrf(request)['csrf_token'],
+            "messages": {
+                "warning": {
+                    "required": _("Are you Sure ?\nIf you do not agree to share your data, you will not get any "
+                                  "discounts from {provider}.").format(provider=name),
+                    "enforced": _("Are you Sure ?\nIf you do not agree to share your data, you will have to use "
+                                  "another account to access {platform}.").format(platform=platform_name)
+                },
+                "note": {
+                    "required": _("*{provider} requires data sharing consent; if consent is not provided, you will"
+                                  " not be able to get any discounts from {provider}.").format(provider=name),
+                    "enforced": _("*{provider} requires data sharing consent; if consent is not provided, you will"
+                                  " be redirected to log in page.").format(provider=name)
```

Comment:
All these messages could use substantial copyedit work, but I'm not worried about that at this point; mainly because I assume Legal et. al. will have some future input.